src/dbsync/src/dbsync.cpp

```diff
@@ -17,78 +19,197 @@ extern "C" {
 using namespace DbSync;
 struct CJsonDeleter
 {
-  void operator()(char* json)
-  {
-    cJSON_free(json);
-  }
+    void operator()(char* json)
+    {
+        cJSON_free(json);
+    }
 };
 
-DBSYNC_HANDLE dbsync_initialize(
-    const HostType host_type, 
-    const DbEngineType db_type,
-    const char* path, 
-    const char* sql_statement) {
+static log_fnc_t gs_logFunction{ nullptr };
 
-  DBSYNC_HANDLE ret_val{ nullptr };
+static void log_message(const std::string& msg)
+{
+    if (!msg.empty())
+    {
+        if (gs_logFunction)
```

Comment:
Please, make the nullptr explicit comparison to maintain the entire file style (see line: 290).